mockwebserver/src/main/java/com/squareup/okhttp/mockwebserver/MockWebServer.java

```diff
@@ -122,17 +132,17 @@ public void setServerSocketFactory(ServerSocketFactory serverSocketFactory) {
   }
 
   public int getPort() {
-    if (port == -1) throw new IllegalStateException("Call play() before getPort()");
+    startOrThrow();
```

Comment:
Yeah, let's go 100% in on the rule. The `play`/`shutdown` names aren't necessary. That comes from an earlier time, when you needed to provide all the responses before you called `play`, and it was intended to offer a record/replay experience.